hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractFSWALProvider.java

```diff
@@ -557,8 +527,9 @@ public static Path findArchivedLog(Path path, Configuration conf) throws IOExcep
         return reader;
       } catch (FileNotFoundException fnfe) {
         // If the log was archived, continue reading from there
-        Path archivedLog = AbstractFSWALProvider.getArchivedLogPath(path, conf);
-        if (!Objects.equals(path, archivedLog)) {
+        Path archivedLog = AbstractFSWALProvider.findArchivedLog(path, conf);
+        // archivedLog can be null if unable to locate in archiveDir.
+        if (archivedLog != null && !Objects.equals(path, archivedLog)) {
```

Comment:
Yeah see my above comment. Here you use Objects.equals which will call through to Path's equals method. This is along the lines of what I proposed, I think you should do this in the other place too.